packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -466,17 +466,71 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
         const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
 
+        let isOptionsCall = false;
+
+        if (Object.keys(headers).length) {
+            isOptionsCall = true;
+        }
+
         // This event measures only successful cases of getLatest call (no tokens, no retries).
         const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
+            const startTime = performance.now();
             const response = await fetchHelper<IOdspSnapshot>(url, { headers });
+            const endTime = performance.now();
+            const overallTime = endTime - startTime;
             const content = response.content;
+            let dnstime = -1; // domainLookupEnd - domainLookupStart
+            let redirectTime = -1; // redirectEnd -redirectStart
+            let tcpHandshakeTime = -1; // connectEnd  - connectStart
+            let secureConntime = -1; // connectEnd  - secureConnectionStart
+            let responseTime = -1; // responsEnd - responseStart
+            let fetchStToRespEndTime = -1; // responseEnd  - fetchStart
```

Comment:
Should we be using undefined instead of -1 as the default value for easier analysis?